kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMConfigurationServiceImpl.java

```diff
@@ -314,6 +318,59 @@ private void writeNetworkConfigurationSettings(Map<String, Object> networkProper
         }
     }
 
+    private void writeFirewallNatRules(Set<String> interfaceNames) {
+        List<String> wanInterfaces = new LinkedList<>();
+        List<String> natInterfaces = new LinkedList<>();
+
+        interfaceNames.forEach(interfaceName -> {
+            if (isWanInterface(interfaceName)) {
+                wanInterfaces.add(interfaceName);
+            }
+        });
+
+        interfaceNames.forEach(interfaceName -> {
+            if (isNatValid(interfaceName)) {
+                natInterfaces.add(interfaceName);
+            }
+        });
+
```

Comment:
Does it make sense to merge these loops?

Reply:
Totally, will make this change soon